### PR TITLE
statistics: merge partition-level FMSketch to global-level FMSketch and update the column NDV

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

sub PR for #22554 

Problem Summary:
merge partition-level FMSketch to global-level FMSketch and update the column NDV

### What is changed and how it works?
Same as described above.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

**Review by qw4990 — COMMENTED** [review]
LGTM

**Review by qw4990 — COMMENTED** [review]
/LGTM

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -409,10 +410,20 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc *stmtctx.StatementContext, i
...
+				globalStats.Fms[i].MergeFMSketch(allFms[i][j])
+			}
+
+			// update the NDV
+			globalStatsNDV := globalStats.Fms[i].NDV()
+			if globalStatsNDV > globalStats.Count {
+				globalStatsNDV = globalStats.Count
+			}
+			globalStats.Hg[i].NDV = globalStatsNDV
 		}
```
Comment (rebelice) [inline statistics/handle/handle.go:427]: Please add an error to warn that index `ndv` is not merged.

**Review by rebelice — COMMENTED** [review]
rest LGTM

File: statistics/handle/handle.go (hunk 1 above)
Comment (rebelice) [reply statistics/handle/handle.go:427]: Or you can calculate the index `ndv` here

File: statistics/handle/handle.go [hunk 2]
```diff
@@ -409,10 +410,30 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc *stmtctx.StatementContext, i
...
+		if isIndex == 0 {
+			// For the column stats, we should merge the FMSketch first. And use the FMSketch to calculate the new NDV.
+			// merge FMSketch
+			globalStats.Fms[i] = allFms[i][0].Copy()
+			for j := uint64(1); j < partitionNum; j++ {
+				globalStats.Fms[i].MergeFMSketch(allFms[i][j])
+			}
+
+			// update the NDV
+			if globalStats.Fms[i] != nil {
```
Comment (Reminiscent) [inline statistics/handle/handle.go:423]: TODO: need more investigation when the `globalStats.Fms[i]` will be nil

Comment (rebelice) [reply statistics/handle/handle.go:423]: Maybe we can return an error simply now.

**Review by rebelice — APPROVED** [review]

**Review by qw4990 — COMMENTED** [review]
LGTM

**Pull request merged**